src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -168,85 +144,57 @@ internal static bool IsExtended(string path)
                 && path[3] == '\\';
         }
 
-        /// <summary>
-        /// Check for known wildcard characters. '*' and '?' are the most common ones.
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static unsafe bool HasWildCardCharacters(string path)
-        {
-            // Question mark is part of dos device syntax so we have to skip if we are
-            int startIndex = PathInternal.IsDevice(path) ? ExtendedPathPrefix.Length : 0;
-
-            return path.IndexOfAny(s_wildcardChars, startIndex) >= 0;
-        }
-
+#if !NOSPAN
         /// <summary>
         /// Gets the length of the root of the path (drive, share, etc.).
         /// </summary>
-        internal static unsafe int GetRootLength(string path)
-        {
-            fixed(char* value = path)
-            {
-                return (int)GetRootLength(value, (uint)path.Length);
-            }
-        }
-
-        private static unsafe uint GetRootLength(char* path, uint pathLength)
+        internal static int GetRootLength(ReadOnlySpan<char> path)
         {
-            uint i = 0;
-            uint volumeSeparatorLength = 2;  // Length to the colon "C:"
-            uint uncRootLength = 2;          // Length to the start of the server name "\\"
+            int i = 0;
```

Comment:
maybe we can use https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/IO/Path.Windows.cs#L275 this function to reduce some checks here ?